From 22ab9bc6fb9723838ad4a6bf4753136cb21b95a4 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Thu, 9 Oct 2014 20:59:56 -0400
Subject: [fs] xfs: extra semi-colon breaks a condition

Message-id: <1412888441-4181-3-git-send-email-sandeen@redhat.com>
Patchwork-id: 97308
O-Subject: [RHEL7.1 PATCH 02/47] [fs] xfs: extra semi-colon breaks a condition
Bugzilla: 1145837
RH-Acked-by: Brian Foster <bfoster@redhat.com>
RH-Acked-by: Dave Chinner <dchinner@redhat.com>
RH-Acked-by: Ian Kent <ikent@redhat.com>

Bugzilla: 1145837
Upstream Status: Committed upstream
Build Info: https://brewweb.devel.redhat.com/taskinfo?taskID=8082426
Tested: locally using xfstests

Thanks,
-Eric

    commit 805eeb8e04706a16cb0b23fd4c4abbb0bc7df82d
    Author: Dan Carpenter <dan.carpenter@oracle.com>
    Date:   Fri Apr 4 06:56:30 2014 +1100

    xfs: extra semi-colon breaks a condition

    There were some extra semi-colons here which mean that we return true
    unintentionally.

    Fixes: a49935f200e2 ('xfs: xfs_check_page_type buffer checks need help')
    Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
    Reviewed-by: Brian Foster <bfoster@redhat.com>
    Reviewed-by: Eric Sandeen <sandeen@redhat.com>
    Signed-off-by: Dave Chinner <david@fromorbit.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/xfs/xfs_aops.c b/fs/xfs/xfs_aops.c
index 1c6b1d5..a30c697 100644
--- a/fs/xfs/xfs_aops.c
+++ b/fs/xfs/xfs_aops.c
@@ -670,10 +670,10 @@ xfs_check_page_type(
    if (type == XFS_IO_UNWRITTEN)
     return true;
   } else if (buffer_delay(bh)) {
-   if (type == XFS_IO_DELALLOC);
+   if (type == XFS_IO_DELALLOC)
     return true;
   } else if (buffer_dirty(bh) && buffer_mapped(bh)) {
-   if (type == XFS_IO_OVERWRITE);
+   if (type == XFS_IO_OVERWRITE)
     return true;
   }
 
-- 
1.7.1